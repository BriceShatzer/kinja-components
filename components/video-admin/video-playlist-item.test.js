/* @flow */

import * as React from 'react';
import { shallow } from 'enzyme';

import SimpleImage from 'kinja-magma/models/SimpleImage';

import VideoPlaylistItem from './video-playlist-item';

describe('<VideoPlaylistItem />', () => {
	const timestamp = 1512136800667;

	const stubbedPoster1 = SimpleImage.fromJSON({
		id: 'igfq6s2z5hywc5yp3xrg',
		format: 'jpg'
	});

	beforeAll(() => {
		jest.spyOn(Date, 'now').mockImplementation(() => timestamp);
	});

	it('should render by default', () => {
		const wrapper = shallow(<VideoPlaylistItem
			index={0}
			count={2}
			videoId='1'
			poster={stubbedPoster1}
			title={'This Video Is Really Good'}
			uploadTime={'2019-10-31T13:30:00'}
			length={260}
			metaSource='Video'
			onRemoveClick={() => {}}
			onMoveItemClick={() => {}}
			onToggleMetaSourceClick={() => {}}
		/>);
		expect(wrapper).toMatchSnapshot();
	});
});
