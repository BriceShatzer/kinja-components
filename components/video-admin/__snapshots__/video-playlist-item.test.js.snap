// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlaylistItem /> should render by default 1`] = `
<VideoPlaylistItem
  count={2}
  dragEvents={null}
  index={0}
  length={260}
  metaSource="Video"
  onMoveItemClick={[Function]}
  onRemoveClick={[Function]}
  onToggleMetaSourceClick={[Function]}
  poster={
    Object {
      "format": "jpg",
      "frozenFormat": "jpg",
      "id": "igfq6s2z5hywc5yp3xrg",
      "isAnimated": false,
    }
  }
  title="This Video Is Really Good"
  uploadTime="2019-10-31T13:30:00"
  videoId="1"
/>
`;
